### ISSUE TITLE
Error response from daemon: Container [id] is not running after entering PiVPN user password

### ISSUE THREAD

**Issue by SeanChristopherConway** [InnovativeInventor/docker-pivpn#2]
I am running the bash script on a clean install of Raspbian Stretch using Docker. For some reason it says the container is not running. When I run docker ps there is no container running. I am fairly new to Docker so perhaps I am doing something wrong, but it is not clear to me if simply running your script and having Docker installed was all I needed to do. Thanks!

**Comment by InnovativeInventor** [comment]
Did you answer the prompts?

**Comment by SeanChristopherConway** [comment]
Yes, everything. Although it seems to hang for a long time when I choose my port.

**Comment by ragefan** [comment]
Hi,
I have got the same issue.
When i check the docker logs, it says : 
standard_init_linux.go:178: exec user process caused "exec format error"

Seems that one of the executable if not from the right format ?

**Comment by InnovativeInventor** [comment]
I'm trying to hunt down the problem, and from this thread:
https://github.com/moby/moby/issues/23865, it looks like when docker isn't
installed (it installs it for you), my script accidentally installed the
non-arm compatible version of docker. I'll comment out the automatic docker
install part of the script while I fix it. Tell me if it works.

**Comment by InnovativeInventor** [comment]
I'm trying to clean up all of my code, so the code will look a lot
different in when I'm done and should be easier to manage.

**Comment by ragefan** [comment]
Well actually, Docker is already installed on my raspberry so i'm not sure the issue is coming from that.
Are you sure of the base image of the Docker container ?

**Comment by InnovativeInventor** [comment]
Yeah. Here is the image that my docker image is based on
FROM ubuntu:16.04
Does ubuntu 16.0.4's docker image work on the Raspberry pi? It does say
that it supports ARM.

**Comment by ragefan** [comment]
No it is not ! Because this is a x64 image. Raspberry pi needs armf images.
You shall use this one i think : 
https://hub.docker.com/r/armv7/armhf-ubuntu/

=> This can explain the exec format error

**Comment by ragefan** [comment]
I tested with a modified DockerFile and i don't have the exec issues anymore.
I can reach the steps up to have the pivpn install interface.
pivpn starts installing and then suddenetly existing while configuring openvpn.
Are you interested in the modified Dockerfile that works with raspberry pi ?

**Comment by InnovativeInventor** [comment]
Yes. That dockerfile sounds like a great idea. When running on ARM, we'll
use the modified docker file, and when running on amd64, we'll use the
current one.

**Comment by InnovativeInventor** [comment]
@ragefan @SeanChristopherConway  I've completely redid the scripts, so they should work. Please tell me if it does.

**Comment by ragefan** [comment]
Hi,
Sorry for my late reply. I just tried again 👍 
Got the following : 
setup.sh: ligne 179: warning :command substitution: ignored null byte in input

I will investigate more

**Comment by ragefan** [comment]
I confirm the code is not working at all on a raspberry pi2.
Would need to restart the whole Docker packaging from another image (i tried debian arm) and even modify pivpn setup scripts to make it work

**Comment by ScooterComputer** [comment]
For some reason, the resin/rpi-raspbian image is no longer working (requires a Docker login?). I modified the Dockerfile and rebuilt (required some changed to your setup.sh, found some bugs), using "FROM resin/raspberry-pi-debian" (which is the image that the rpi-raspbian is aliased to). That seemed to work. 
Though, going forward, Resin is changing/breaking the armhf tag, instead using the tags armv6hf and armv7hf. Now Pi2 and Pi3 use a different Resin base image (armv7hf) than Pi1/0.

**Comment by InnovativeInventor** [comment]
Great! Pull requests are always welcome!

**Comment by InnovativeInventor** [comment]
It looks like armv7hf might be backwards-compatible with armv6hf, although I'm not 100% sure about that.
